src/nifreeze/viz/motion_viz.py

```diff
@@ -0,0 +1,267 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+#
+# Copyright The NiPreps Developers <nipreps@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.axes import Axes
+from scipy.ndimage import gaussian_filter
+
+
+def _get_axis(orientation: str):
+    """Get axis for the given orientation.
+
+    Parameters
+    ----------
+    orientation : :obj:`str`
+        Orientation. Cn be one of ``axial``, ``coronal``, or ``sagittal``.
+
+    Returns
+    -------
+    :obj:`int`
+        Axis corresponding to the given orientation.
+    """
+
+    if orientation == "axial":
+        return 2
+    elif orientation == "coronal":
+        return 1
+    elif orientation == "sagittal":
+        return 0
+    else:
+        raise ValueError(
+            f"Orientation must be one of 'axial', 'coronal', 'sagittal'. Provided: {orientation}"
+        )
+
+
+def _extract_slice(img_data: np.ndarray, orientation: str, slice_idx: int):
+    """Extract slice data from the given volume at the given orientation slice index.
+
+    Parameters
+    ----------
+    img_data : :obj:`~numpy.ndarray`
+        Image data to be sliced.
+    orientation : :obj:`str`
+        Orientation.
+    slice_idx : :obj:`int`
+        Slice index.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Image slice.
+    """
+
+    axis = _get_axis(orientation)
+
+    axis_sizw = img_data.shape[axis]
+
+    if not (0 <= slice_idx < axis_sizw):
+        raise IndexError(
+            f"Slice index {slice_idx} out of bounds for axis {orientation} with size {axis_sizw}"
+        )
+
+    if axis == 2:
+        return img_data[:, :, slice_idx]
+    elif axis == 1:
+        return np.rot90(img_data[:, slice_idx, :])
+    else:
+        return np.rot90(img_data[slice_idx, :, :])
+
+
+def plot_framewise_displacement(
+    fd1: np.ndarray,
+    fd2: np.ndarray,
+    label1: str,
+    label2: str,
```

Comment:
Can we make this method take a pandas dataframe instead (so the column names are kept and used as labels) so we can plot an arbitrary number of compared FD traces?

Otherwise, without pandas (also good option IMHO) we can make the FD be an TxN (N=number of FD series, T=time) matrix.

```suggestion
    fd_series: np.ndarray,
    columns: list[str],
```

Reply:
Done in e8fde0c. The input is now a pandas dataframe.